### PR TITLE
Add OUI to sc open txn

### PR DESCRIPTION
**Pull request by vihu**
* Need to supply `oui` for creating sc open txns
* Remove nonce check from sc open is_valid
* Add tests for newly added checks to sc open is_valid

### CODE REVIEW

File: src/transactions/v1/blockchain_txn_state_channel_open_v1.erl [hunk 1]
```diff
@@ -148,6 +130,53 @@ print(#blockchain_txn_state_channel_open_v1_pb{id=ID, owner=Owner, expire_within
...
+    OUI = ?MODULE:oui(Txn),
+    case ExpireWithin > ?MIN_EXPIRE_WITHIN andalso ExpireWithin < ?APPROX_BLOCKS_IN_WEEK of
+        false ->
+            {error, invalid_expire_at_block};
+        true ->
+            case blockchain_ledger_v1:find_routing(OUI, Ledger) of
+                {error, not_found} ->
+                    {error, {not_found, OUI, Owner}};
+                {ok, Routing} ->
+                    case blockchain_ledger_routing_v1:oui(Routing) == OUI of
```
Comment (Vagabond) [inline src/transactions/v1/blockchain_txn_state_channel_open_v1.erl:147]: You need to check the 'owner' is a router in this OUI as well

Comment (vihu) [reply src/transactions/v1/blockchain_txn_state_channel_open_v1.erl]: Added

File: src/transactions/v1/blockchain_txn_state_channel_open_v1.erl [hunk 2]
```diff
@@ -148,6 +130,57 @@ print(#blockchain_txn_state_channel_open_v1_pb{id=ID, owner=Owner, expire_within
...
+            case blockchain_ledger_v1:find_routing(OUI, Ledger) of
+                {error, not_found} ->
+                    lager:error("oui: ~p not found for this router: ~p", [OUI, Owner]),
+                    {error, {not_found, OUI, Owner}};
+                {ok, Routing} ->
+                    KnownRouters = blockchain_ledger_routing_v1:addresses(Routing),
+                    case lists:member(Owner, KnownRouters) of
+                        false ->
+                            lager:error("unknown router: ~p, known routers: ~p", [Owner, KnownRouters]),
+                            {error, unknown_owner};
```
Comment (Vagabond) [inline src/transactions/v1/blockchain_txn_state_channel_open_v1.erl:152]:
```suggestion
                            {error, unknown_router};
```

File: src/transactions/v1/blockchain_txn_state_channel_open_v1.erl [hunk 3]
```diff
@@ -148,6 +130,57 @@ print(#blockchain_txn_state_channel_open_v1_pb{id=ID, owner=Owner, expire_within
...
+                                        {error, not_found} ->
+                                            case blockchain_ledger_v1:find_dc_entry(Owner, Ledger) of
+                                                {error, _}=Err0 ->
+                                                    Err0;
+                                                {ok, DCEntry} ->
+                                                    TxnNonce = ?MODULE:nonce(Txn),
+                                                    NextLedgerNonce = blockchain_ledger_data_credits_entry_v1:nonce(DCEntry) + 1,
+                                                    case TxnNonce =:= NextLedgerNonce of
+                                                        false ->
+                                                            {error, {bad_nonce, {state_channel_open, TxnNonce, NextLedgerNonce}}};
```
Comment (Vagabond) [inline src/transactions/v1/blockchain_txn_state_channel_open_v1.erl:169]: We mostly do nonce checks in absorb, should this move there?

Comment (vihu) [reply src/transactions/v1/blockchain_txn_state_channel_open_v1.erl:169]: Hmm, `absorb` does `debit_dc` which does a nonce check. Wonder if it's okay to keep this here to discard the txn early?

Comment (vihu) [reply src/transactions/v1/blockchain_txn_state_channel_open_v1.erl]: Added 658034b

**Pull request merged**